sql/2019/08_Security/08_29.sql

```diff
@@ -1,22 +1,31 @@
 #standardSQL
-# 08_29: Groupings of "x-content-type-options" parsed values by percentage
-#    
-#
-#   `httparchive.summary_requests.2019_07_01_*` = 118.3 GB
+# 08_29: Groupings of "x-content-type-options" values
+CREATE TEMPORARY FUNCTION extractHeader(payload STRING, name STRING)
+RETURNS STRING LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(payload);
+  var header = $._headers.response.find(h => h.toLowerCase().startsWith(name.toLowerCase()));
+  if (!header) {
+    return null;
+  }
+  return header.split(':')[1].trim();
+} catch (e) {
+  return null;
+}
+''';
 
 SELECT
-  _TABLE_SUFFIX AS client,
-  policy,
+  client,
+  NORMALIZE_AND_CASEFOLD(extractHeader(payload, 'X-Content-Type-Options')) AS value,
```

Comment:
Lot of interesting unicode characters get used here, like , huh... How'd you think to look for this?

Reply:
Could have used `LOWER`. Normalization may be overkill.